packages/cli/src/credentials/credentials.service.ts

```diff
@@ -602,4 +602,25 @@ export class CredentialsService {
 			mergedCredentials.data = decryptedData;
 		}
 	}
+
+	/**
+	 * Create a new credential in user's account and return it along the scopes
+	 * If a projectId is send, then it also binds the credential to that specific project
+	 */
+	async createCredential(credentialsData: CredentialRequest.CredentialProperties, user: User) {
```

Comment:
Suggestion/optional - Would be good to set the return type here that captures both credentials and scopes for clarity.